runtime/compiler/control/CompilationThread.cpp

```diff
@@ -2618,6 +2618,10 @@ void TR::CompilationInfo::prepareForCheckpoint()
    J9JavaVM   *vm       = _jitConfig->javaVM;
    J9VMThread *vmThread = vm->internalVMFunctions->currentVMThread(vm);
 
+   bool hadVMAccess = (vmThread->publicFlags & J9_PUBLIC_FLAGS_VM_ACCESS);
+   if (hadVMAccess)
+      releaseVMAccessNoSuspend(vmThread);
+
    {
    OMR::CriticalSection suspendCompThreadsForCheckpoint(getCompilationMonitor());
 
```

Comment:
The early return below makes this routine to exit without VMAccess (after possibily entering with VMAccess)

Reply:
Good catch, that'd be pretty bad, especially because it would be happening in a very rare case (shutdown or something similar).